src/components/Users/Users.js

```diff
@@ -1,12 +1,30 @@
-import User from '../User/User';
-import './Users.css';
+import User from "../User/User";
+import "./Users.css";
+
+const Users = ({ users, setExpanded, expanded }) => {  
+  const handleExpanded = (id) => {
+    let copyExpaned = expanded;
+
+    if (copyExpaned.includes(id)) {
+      setExpanded(copyExpaned.filter((userId) => userId !== id));
+    } else {
+      setExpanded([...copyExpaned, id]);
+    }
+  };
 
-const Users = ({ users = [] }) => {
   return (
     <article className="Users">
       {users.map((user) => {
         const { id } = user;
-        return <User key={id} user={user} />;
+        return (
+          <User
+            key={id}
+            user={user}
+            handleExpanded={handleExpanded}
+            expanded={expanded}
```

Comment:
Passing the entire `expanded` array to a single `<User />` works in this implementation because you have the User check if its id is in the array. However, this is not the approach I would take personally. I would rather do that check for `expanded.includes(id)` here in `<Users />` and pass a simple boolean prop to `<User />`.

The reason why I prefer this approach is because the main idea of lifting state up is to give total control and responsibility for the state back to the parent. This means it's the parent's responsibility to figure out if the child should be expanded. In the code here, the parent is passing the entire array down to the child and making the child responsible for figuring it out. I think it would be better for the parent to take full control for this.

In addition, it's just a good idea in software design in general to make sure that each part of the system has a single, clearly defined responsibility. Another way of looking at that is: for all of the things that the system has to do, we want there to be only one part of the system that is fully responsible for it. When you design your app like this, it will be much easier to debug, refactor and extend its functionality over time.